sql/core/src/main/scala/org/apache/spark/sql/expressions/ReduceAggregator.scala

```diff
@@ -32,7 +32,10 @@ private[sql] class ReduceAggregator[T: Encoder](func: (T, T) => T)
 
   @transient private val encoder = implicitly[Encoder[T]]
 
-  override def zero: (Boolean, T) = (false, null.asInstanceOf[T])
```

Comment:
Sorry I may miss the point. Why it's a problem only for spark connect?

Reply:
@cloud-fan I introduced a new code path only in Spark Connect where we do reduce with agg operator with the original type T of the dataset. Previously you can either only do agg + groupByKey with T or agg + Row. In these two paths, there are no null partitions to the reduce aggregators. So this is only caught by spark connect.

Reply:
That said, `null.asInstanceOf[T]` is wrong but it was never triggered before Spark Connect?

Reply:
@zhenlineo and I looked at this last week. The reason why it is not a problem for the old API is because that uses the RDD reduce API instead of SQL aggregation. The RDD API has builtin code that ignores empty partitions.